File: 찬미/Greedy/1931.swift
//
//  1931.swift
//  algorithm
//
//  Created by 황찬미 on 2022/05/15.
//

import Foundation

let number = Int(readLine()!)!
var times: [(Int, Int)] = []

for _ in 1...number {
    let input = readLine()!.split(separator: " ").map {Int($0)!}
    
    times.append((input[0], input[1]))
}

// 시작하는 시간을 기준으로 정렬을 한 후, 끝나는 시간을 기준으로 정렬을 하는 로직 사용...
times.sort(by: { $0.0 < $1.0 }) // 시작하는 시간 기준 오름차순
times.sort(by: { $0.1 < $1.1 }) // 끝나는 시간 기준 오름차순

var endTime = 0
var cnt = 0

for time in times {
    if time.0 >= endTime {
        endTime = time.1
        cnt += 1
    }
}

print(cnt)

File: 윤서/그리디/1931.py
# 1932 - 회의실 배정

import sys
input = sys.stdin.readline

n = int(input().split())            # 입력받을 회의 개수
conf = [[]*2 for _ in range(n)]     # 회의 입력
for i in range(n) :
    start, end = map(int, input().split())      # 회의의 시작시간과 종료 시간 저장
    conf[i][0] = start
    conf[i][1] = end

conf.sort(key = lambda x : (x[1, x[0]]))

count = 1
endTime = conf[0][1]

for i in range(1, n) :                      # 첫번째 회의는 제일 먼저 시작, 볼 필요 없음
    if conf[i][0] >= endTime :              # 지금 회의가 끝나는 시간보다 다음회의의 시작시간이 늦으면
        count += 1
        endTime = conf[i][1]
        
print(count)
 

